src/wallet/wallet2.cpp

```diff
@@ -5407,20 +5732,37 @@ void wallet2::get_outs(std::vector<std::vector<tools::wallet2::get_outs_entry>>
       size_t requested_outputs_count = base_requested_outputs_count + (td.is_rct() ? CRYPTONOTE_MINED_MONEY_UNLOCK_WINDOW - CRYPTONOTE_DEFAULT_TX_SPENDABLE_AGE : 0);
       size_t start = req.outputs.size();
 
-      // if there are just enough outputs to mix with, use all of them.
-      // Eventually this should become impossible.
       uint64_t num_outs = 0, num_recent_outs = 0;
-      for (const auto &he: resp_t.result.histogram)
+      if (m_segregate_pre_fork_outputs && td.m_block_height < segregation_fork_height)
+      {
+        num_outs = segregation_limit[amount].first;
+        num_recent_outs = segregation_limit[amount].second;
+      }
+      else
```

Comment:
After trying this, it makes the comment confusing since it's either after the code, or inside the end of the previous block. Since the indent change is small, I'm leaving it as is.

Reply:
OK, no big deal anyway.